src/Raven.Server/Documents/AttachmentsStorage.cs

```diff
@@ -166,6 +166,11 @@ public IEnumerable<DynamicJsonValue> GetAttachmentsMetadataForDocumentWithCounts
                 if (TableValueToFlags((int)DocumentsTable.Flags, ref tvr).HasFlag(DocumentFlags.Artificial))
                     throw new InvalidOperationException($"Cannot put attachment {name} on artificial document '{documentId}'.");
 
+                if (documentId.Length + name.Length > DocumentIdWorker.MaxIdSize)
```

Comment:
Can we check this earlier? After checking if transaction is null?

Reply:
done